lib/dor/models/releaseable.rb

```diff
@@ -79,7 +79,7 @@ def get_self_release_tags(tags)
     def get_release_tags_for_item_and_all_governing_sets
       return_tags = release_nodes || {}
       collections.each do |collection|
-        return_tags = combine_two_release_tag_hashes(return_tags, Dor::Item.find(collection.id).get_release_tags_for_item_and_all_governing_sets) # recurvise so parents of parents are found
+        return_tags = combine_two_release_tag_hashes(return_tags, Dor::Item.find(collection.id).get_release_tags_for_item_and_all_governing_sets) unless collection.id == id # recurvise so parents of parents are found, but avoid an infinite loop if the collection references itself
```

Comment:
point #2.. ran into this as well. Is it a by design thing in DorServices to not use any classes besides the main classes? `Dor::Item`, etc.. 

I found it really difficult to work with this module because everything returned a special data structure hash or array thingy